src/main/java/uk/gov/ch/developer/docs/interceptor/UserDetailsInterceptor.java

```diff
@@ -0,0 +1,44 @@
+package uk.gov.ch.developer.docs.interceptor;
+
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.lang.Nullable;
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.ModelAndView;
+import org.springframework.web.servlet.handler.HandlerInterceptorAdapter;
+import org.springframework.web.servlet.view.UrlBasedViewResolver;
+import uk.gov.ch.developer.docs.session.SessionService;
+
+@Component
+public class UserDetailsInterceptor extends HandlerInterceptorAdapter {
+
+    private static final String USER_EMAIL = "userEmail";
+    private static final String SIGN_IN_KEY = "signin_info";
+    private static final String USER_PROFILE_KEY = "user_profile";
+    private static final String EMAIL_KEY = "email";
+
+    @Autowired
+    private SessionService sessionService;
+
+    @Override
+    public void postHandle(HttpServletRequest request, HttpServletResponse response, Object handler,
+            @Nullable ModelAndView modelAndView) throws Exception {
+
+        if (modelAndView != null && modelAndView.getViewName() != null
+                && (request.getMethod().equalsIgnoreCase("GET")
+                || (request.getMethod().equalsIgnoreCase("POST")
+                && !modelAndView.getViewName()
+                .startsWith(UrlBasedViewResolver.REDIRECT_URL_PREFIX)))) {
+
+            Map<String, Object> sessionData = sessionService.getSessionDataFromContext();
+            Map<String, Object> signInInfo = (Map<String, Object>) sessionData.get(SIGN_IN_KEY);
+            if (signInInfo != null) {
+                Map<String, Object> userProfile = (Map<String, Object>) signInInfo
+                        .get(USER_PROFILE_KEY);
+                modelAndView.addObject(USER_EMAIL, userProfile.get(EMAIL_KEY));
+            }
```

Comment:
Aren't these logically better handled through the methods on the `HttpServletRequest`, such as:

`String getRemoteUser()`
Returns the login of the user making this request, if the user has been authenticated, or null if the user has not been authenticated. Whether the user name is sent with each subsequent request depends on the browser and type of authentication. Same as the value of the CGI variable REMOTE_USER.
Returns:
a String specifying the login of the user making this request, or null if the user login is not known

`java.security.Principal getUserPrincipal()`
Returns a java.security.Principal object containing the name of the current authenticated user. If the user has not been authenticated, the method returns null.
Returns:
a java.security.Principal containing the name of the user making this request; null if the user has not been authenticated

Reply:
Since this doesn't really have anything to do with security I don't really mind.
User Profile and SignInInfo here are ch classes not standard Java so I'm not sure they have those methods.